users/serializers.py

```diff
@@ -0,0 +1,51 @@
+from rest_framework import serializers
+from .models import UserProfile
+from django.utils.translation import ugettext_lazy
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserProfile
+        fields = ['username', 'email']
+
+
+class CreateUserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserProfile
+        fields = ['id', 'username', 'email', 'password']
+        read_only_fields = ['id']
+        extra_kwargs = {'password': {'write_only': True}}
+
+    def create(self, validated_data):
+        password = validated_data['password']
+        user = UserProfile(username=validated_data['username'], email=validated_data['email'])
+        user.set_password(password)
+        user.save()
+        return user
+
+
+class CustomAuthTokenSerializer(serializers.Serializer):
+    email = serializers.EmailField()
+    password = serializers.CharField()
+
+    def validate(self, attrs):
+        email = attrs.get('email')
+        password = attrs.get('password')
+
+        if email and password:
+            user = UserProfile.objects.get(email=email)
```

Comment:
It raises an error if an user does not exist with the given email. You need to handle this error.